viewer/packages/camera-manager/src/CameraManager.ts

```diff
@@ -1,604 +1,75 @@
-  ): THREE.Vector3 {
-    const modelSize = modelsBoundingBox.min.distanceTo(modelsBoundingBox.max);
-
-    const lastScrollTargetDistance = this.cameraControls.getScrollTarget().distanceTo(this._camera.position);
-
-    const newTargetDistance =
-      lastScrollTargetDistance <= this.cameraControls.minDistance
-        ? Math.min(this._camera.position.distanceTo(modelsBoundingBox.getCenter(new THREE.Vector3())), modelSize) / 2
-        : lastScrollTargetDistance;
-
-    this._raycaster.setFromCamera(cursorPosition, this._camera);
-
-    const farPoint = this._raycaster.ray.direction
-      .clone()
-      .normalize()
-      .multiplyScalar(newTargetDistance)
-      .add(this._camera.position);
-
-    return farPoint;
-  }
+  setCameraState(state: CameraState): void;
 
   /**
-   * Calculates new camera target based on cursor position.
-   * @param event MouseEvent that contains pointer location data.
+   * Get camera's state
+   * @returns Camera state: position, target and rotation.
    */
-  private async calculateNewTarget(event: MouseEvent): Promise<THREE.Vector3> {
-    const { offsetX, offsetY } = event;
-    const { x, y } = this.convertPixelCoordinatesToNormalized(offsetX, offsetY);
-
-    const modelRaycastData = await this._modelRaycastCallback(offsetX, offsetY);
-
-    const newTarget =
-      modelRaycastData.intersection?.point ??
-      this.calculateNewTargetWithoutModel({ x, y }, modelRaycastData.modelsBoundingBox);
-
-    return newTarget;
-  }
+  getCameraState(): Required<CameraState>;
 
   /**
-   * Removes controls event listeners if they are defined.
+   * Subscribes to changes of the camera event. This is used by Reveal to react on changes of the camera.
+   * @param event Name of the event.
+   * @param callback Callback to be called when the event is fired.
    */
-  private teardownControls(removeOnWheel: boolean = true): void {
-    if (this._onClick !== undefined) {
-      this._inputHandler.off('click', this._onClick as PointerEventDelegate);
-      this._onClick = undefined;
-    }
-    if (this._onWheel !== undefined && removeOnWheel) {
-      this._domElement.removeEventListener('wheel', this._onWheel);
-      this._onWheel = undefined;
-    }
-  }
-
-  private handleMouseWheelActionChange(controlsOptions: CameraControlsOptions) {
-    const { _controls: controls } = this;
-
-    switch (controlsOptions?.mouseWheelAction) {
-      case 'zoomToTarget':
-        controls.zoomToCursor = false;
-        break;
-      case 'zoomPastCursor':
-        controls.useScrollTarget = false;
-        controls.zoomToCursor = true;
-        break;
-      case 'zoomToCursor':
-        controls.setScrollTarget(controls.getState().target);
-        controls.useScrollTarget = true;
-        controls.zoomToCursor = true;
-        break;
-      case undefined:
-        break;
-      default:
-        assertNever(controlsOptions.mouseWheelAction);
-    }
-  }
+  on(event: 'cameraChange', callback: CameraChangeData): void;
```

Comment:
Huh, is `CameraChangeData` the callback definition? Or the data in the event?

I think it should be renamed to `CameraChangedEvent`

Reply:
Renamed